ui/src/treeDataProvider/AzureTreeDataProvider.ts

```diff
@@ -220,6 +221,25 @@ export class AzureTreeDataProvider<TRoot = ISubscriptionRoot> implements IAzureT
         return element.parent;
     }
 
+    public async getSubscriptionPromptStep(wizardContext: Partial<types.ISubscriptionWizardContext>): Promise<types.AzureWizardPromptStep<types.ISubscriptionWizardContext> | undefined> {
```

Comment:
Why aren't you creating a .ts file that is the SubscriptionPromptStep rather than implementing it in the `AzureTreeDataProvider.ts`?

Reply:
Just a stylistic preference. It's not that many lines of code and it should only be used within this function anyways